var/spack/repos/builtin/packages/llvm-flang/package.py

```diff
@@ -0,0 +1,168 @@
+
+    resource(name='flang-driver',
+             git='https://github.com/flang-compiler/flang-driver.git',
+             tag='flang_20181226_70',
+             destination='tools',
+             placement='clang',
+             when='@20181226_70')
+
+    resource(name='flang-driver',
+             git='https://github.com/flang-compiler/flang-driver.git',
+             tag='flang_20181226_60',
+             destination='tools',
+             placement='clang',
+             when='@20181226_60')
+
+    resource(name='flang-driver',
+             git='https://github.com/flang-compiler/flang-driver.git',
+             tag='flang_20180921',
+             destination='tools',
+             placement='clang',
+             when='@20180921')
+
+    resource(name='flang-driver',
+             git='https://github.com/flang-compiler/flang-driver.git',
+             tag='flang_20180921',
+             destination='tools',
+             placement='clang',
+             when='@20180308')
+
+    resource(name='openmp',
+             git='https://github.com/flang-compiler/openmp.git',
+             branch='master',
+             destination='projects',
+             placement='openmp',
+             when='@master')
+
+    resource(name='openmp',
+             git='https://github.com/flang-compiler/openmp.git',
+             branch='release_70',
+             destination='projects',
+             placement='openmp',
+             when='@release_70')
+
+    resource(name='openmp',
+             git='https://github.com/flang-compiler/openmp.git',
+             branch='release_60',
+             destination='projects',
+             placement='openmp',
+             when='@release_60')
+
+    resource(name='openmp',
+             git='https://github.com/flang-compiler/openmp.git',
+             tag='flang_20190329',
+             destination='projects',
+             placement='openmp',
+             when='@20190329')
+
+    resource(name='openmp',
+             git='https://github.com/flang-compiler/openmp.git',
+             tag='flang_20181226_70',
+             destination='projects',
+             placement='openmp',
+             when='@20181226_70')
+
+    resource(name='openmp',
+             git='https://github.com/flang-compiler/openmp.git',
+             tag='flang_20181226_60',
+             destination='projects',
+             placement='openmp',
+             when='@20181226_60')
+
+    resource(name='openmp',
+             git='https://github.com/flang-compiler/openmp.git',
+             tag='flang_20180921',
+             destination='projects',
+             placement='openmp',
+             when='@20180921')
+
+    def cmake_args(self):
+        spec = self.spec
+        args = []
+        args.append('-DPYTHON_EXECUTABLE={0}'.format(
+            spec['python'].command.path))
+
+        if '+all_targets' not in spec:  # all is default in cmake
+            if spec.target.family == 'x86' or spec.target.family == 'x86_64':
```

Comment:
I would put `target = ''` before this line just in case someone is running on a platform not in this list.

Reply:
I think it would be better to throw a hard error if its not one of the options on the list. What
is the best way to do that? Based on my perfunctory review of CMakelists.txt, it is not clear what CMake would do if you passed an empty string to                 `-DLLVM_TARGETS_TO_BUILD:STRING=`

I supposed I could hack it to fallback to `x86` or `all` -- but I think it would be better to throw a hard error.

I also think that I need to go back and remove 'Sparc' -- does not look like its actually supported.

Reply:
I would do something like:
```python
raise InstallError('Unsupported architecture: ' + spec.target.family)